src/Entity/Character.cpp

```diff
@@ -130,5 +134,38 @@ void CharacterUpdate(flecs::entity characterEntity, TransformComponent& transfor
 
 }
 
-}// namespace character	
-}// namespace game
+void SpawnCoin(flecs::world& ecs, const Vector2 position) {
+
+	static int index = 0;
+	const std::string coinName = std::string("coin" + std::to_string(index++));
```

Comment:
The constructor call is not needed since the `+` operator is overloaded to return a `std::string`, so it can simply be:
```cpp
	const std::string coinName = "coin" + std::to_string(index++);
```